### PR TITLE
PCM-1972, PCM-1789 - Connect WebRTC data channel, listen/emit for active speaker update messages.

### PR DESCRIPTION
**Pull request by zservies**

### CODE REVIEW

File: src/types/interfaces.ts [hunk 1]
```diff
@@ -1163,3 +1164,51 @@ export interface IVideoResolution {
...
+export interface IMemberStatusMessage extends JsonRpcMessage {
+  method: 'member.notify.status';
+  params: {
+    speakers?: VideoSpeakerStatus[];
+    outgoingStreams?: OutgoingStreamStatus[];
+    incomingStreams?: IncomingStreamStatus[];
+  }
+}
+
+export interface VideoSpeakerStatus {
```
[the remark this answers is not in the document]
Comment (zservies) [reply src/types/interfaces.ts:1177]: Not entirely, I need to update it. There is an `id`, although its not the `userId`, I'm not sure what it actually is. There's another field I need to add to this:
```
appId: {
 sourceJid: string,
 sourceParticipantId: string,
 sourceUserId: string //This is the userId we look for in volt
 }
 ```

[the remark this answers is not in the document]
Comment (zservies) [reply src/types/interfaces.ts:1177]: Updated.

File: src/types/interfaces.ts [hunk 2]
```diff
@@ -1163,3 +1164,63 @@ export interface IVideoResolution {
...
+  id: string;
+
+  activity: 'speaking' | 'inactive' | 'non-speech';
```
Comment (zservies) [inline src/types/interfaces.ts:1180]: This _may_ need to be updated after Kevin starts sending messages to indicate a user has stopped speaking, I don't know what that `activity` will look like.

**Review by maxwellmooney13 — APPROVED** [review]

**Comment by jensengar** [comment]
Can we squash all the commits?

**Comment by jensengar** [comment]
> Collaborator

Other than this, I'm good with this pr once we get the streaming-client released and bumped.

**Comment by jensengar** [comment]
Other than this, I'm good with this pr once we get the streaming-client released and bumped.

**Pull request merged**